src/pytest_html/extras.py

```diff
@@ -20,41 +28,51 @@ def extra(content, format_type, name=None, mime_type=None, extension=None):
     }
 
```

Comment:
`html()`, `image()` etc are facades to `extra()`.

to make this more clear, i suggest to define a variable with the return type of `extra()`  as `_ExtraResult = Dict[str, Optional[str]]`
and use this `_ExtraResult` where needed for `extra()`, `html()` and so on